flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -315,15 +320,28 @@ public DynamicTableSource createDynamicTableSource(Context context) {
 	public DynamicTableSink createDynamicTableSink(Context context) {
 		FactoryUtil.TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
 		helper.validate();
+		String sinkClass = helper.getOptions().get(TABLE_SINK_CLASS);
+
 		boolean isInsertOnly = helper.getOptions().get(SINK_INSERT_ONLY);
 		String runtimeSink = helper.getOptions().get(RUNTIME_SINK);
 		int expectedNum = helper.getOptions().get(SINK_EXPECTED_MESSAGES_NUM);
 		TableSchema schema = context.getCatalogTable().getSchema();
-		return new TestValuesTableSink(
-			schema,
-			context.getObjectIdentifier().getObjectName(),
-			isInsertOnly,
-			runtimeSink, expectedNum);
+		if (sinkClass.equals("DEFAULT")) {
+			return new TestValuesTableSink(
+				schema,
+				context.getObjectIdentifier().getObjectName(),
+				isInsertOnly,
+				runtimeSink, expectedNum);
+		} else {
+			try {
+				return InstantiationUtil.instantiate(
+					sinkClass,
+					DynamicTableSink.class,
+					Thread.currentThread().getContextClassLoader());
+			} catch (FlinkException e) {
```

Comment:
Also use `TableException` ?

Reply:
I think wee can't use `TableException` here, because `InstantiationUtil.instantiate` throws `FlinkException` which is defined in `flink-core`.

Reply:
got it